sherpa/astro/data.py

```diff
@@ -2541,10 +2601,17 @@ def _get_ebins(self,
         if self.size is None:
             raise DataErr("sizenotset", self.name)
 
+        # This is primarily to make the type checker happy, since it
+        # is expected to be called with channels set.
+        #
+        if self.channel is None:
+            raise DataErr(f"data set '{self.name}' has no channel information")
+
         if self.units == 'channel':
             elo = self.channel
             ehi = self.channel + 1
         elif (self.bin_lo is not None) and (self.bin_hi is not None):
+            # TODO: review whether bin_lo/hi should over-ride the response
```

Comment:
We should just remove bin_hi / bin_lo and always require a response. But not in this PR.

Reply:
For completeness, I will link to 

- #1564 
- #1871 

as it can be useful to see what the discussions are.